File: src/Model/Maintenance.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Model;
import java.util.ArrayList;
import java.util.List;
import java.util.*;
/**
 *
 * @author helam
 */
public class Maintenance {
    //attributs
    private int id_maintenance;
    private User user;
    private Date date_maintenance;
    private boolean panne_moteur,pompe_a_eau,patin,essuie_glace,radiateur,ventilateur,duride,fuite_d_huile,vidange,filtre,batterie,amortisseur,frein_main,feu_d_eclairage;
    private String Autre;

   
   ///constructeurs

    public Maintenance() {
    }

    public Maintenance(int id_maintenance, User user, Date date_maintenance, boolean panne_moteur, boolean pompe_a_eau, boolean patin, boolean essuie_glace, boolean radiateur, boolean ventilateur, boolean duride, boolean fuite_d_huile, boolean vidange, boolean filtre, boolean batterie, boolean amortisseur, boolean frein_main, boolean feu_d_eclairage, String Autre) {
        this.id_maintenance = id_maintenance;
        this.user = user;
        this.date_maintenance = date_maintenance;
        this.panne_moteur = panne_moteur;
        this.pompe_a_eau = pompe_a_eau;
        this.patin = patin;
        this.essuie_glace = essuie_glace;
        this.radiateur = radiateur;
        this.ventilateur = ventilateur;
        this.duride = duride;
        this.fuite_d_huile = fuite_d_huile;
        this.vidange = vidange;
        this.filtre = filtre;
        this.batterie = batterie;
        this.amortisseur = amortisseur;
        this.frein_main = frein_main;
        this.feu_d_eclairage = feu_d_eclairage;
        this.Autre = Autre;
    }

    public Maintenance(User user, boolean panne_moteur, boolean pompe_a_eau, boolean patin, boolean essuie_glace, boolean radiateur, boolean ventilateur, boolean duride, boolean fuite_d_huile, boolean vidange, boolean filtre, boolean batterie, boolean amortisseur, boolean frein_main, boolean feu_d_eclairage, String Autre) {
        this.user = user;
        this.panne_moteur = panne_moteur;
        this.pompe_a_eau = pompe_a_eau;
        this.patin = patin;
        this.essuie_glace = essuie_glace;
        this.radiateur = radiateur;
        this.ventilateur = ventilateur;
        this.duride = duride;
        this.fuite_d_huile = fuite_d_huile;
        this.vidange = vidange;
        this.filtre = filtre;
        this.batterie = batterie;
        this.amortisseur = amortisseur;
        this.frein_main = frein_main;
        this.feu_d_eclairage = feu_d_eclairage;
        this.Autre = Autre;
    }

    
   
   
   
   
   ///toString 
    @Override
    public String toString() {
        return "Maintenance{" + ", id_user=" + user + ", date_maintenance=" + date_maintenance + ", panne_moteur=" + panne_moteur + ", pompe_a_eau=" + pompe_a_eau + ", patin=" + patin + ", essuie_glace=" + essuie_glace + ", radiateur=" + radiateur + ", ventilateur=" + ventilateur + ", duride=" + duride + ", fuite_d_huile=" + fuite_d_huile + ", vidange=" + vidange + ", filtre=" + filtre + ", batterie=" + batterie + ", amortisseur=" + amortisseur + ", frein_main=" + frein_main + ", feu_d_eclairage=" + feu_d_eclairage + ", Autre=" + Autre + '}';
    }

    ///getters & setters
    public int getId_maintenance() {
        return id_maintenance;
    }

    public void setId_maintenance(int id_maintenance) {
        this.id_maintenance = id_maintenance;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Date getDate_maintenance() {
        return date_maintenance;
    }

    public void setDate_maintenance(Date date_maintenance) {
        this.date_maintenance = date_maintenance;
    }

    public boolean isPanne_moteur() {
        return panne_moteur;
    }

    public void setPanne_moteur(boolean panne_moteur) {
        this.panne_moteur = panne_moteur;
    }

    public boolean isPompe_a_eau() {
        return pompe_a_eau;
    }

    public void setPompe_a_eau(boolean pompe_a_eau) {
        this.pompe_a_eau = pompe_a_eau;
    }

    public boolean isPatin() {
        return patin;
    }

    public void setPatin(boolean patin) {
        this.patin = patin;
    }

    public boolean isEssuie_glace() {
        return essuie_glace;
    }

    public void setEssuie_glace(boolean essuie_glace) {
        this.essuie_glace = essuie_glace;
    }

    public boolean isRadiateur() {
        return radiateur;
    }

    public void setRadiateur(boolean radiateur) {
        this.radiateur = radiateur;
    }

    public boolean isVentilateur() {
        return ventilateur;
    }

    public void setVentilateur(boolean ventilateur) {
        this.ventilateur = ventilateur;
    }

    public boolean isDuride() {
        return duride;
    }

    public void setDuride(boolean duride) {
        this.duride = duride;
    }

    public boolean isFuite_d_huile() {
        return fuite_d_huile;
    }

    public void setFuite_d_huile(boolean fuite_d_huile) {
        this.fuite_d_huile = fuite_d_huile;
    }

    public boolean isVidange() {
        return vidange;
    }

    public void setVidange(boolean vidange) {
        this.vidange = vidange;
    }

    public boolean isFiltre() {
        return filtre;
    }

    public void setFiltre(boolean filtre) {
        this.filtre = filtre;
    }

    public boolean isBatterie() {
        return batterie;
    }

    public void setBatterie(boolean batterie) {
        this.batterie = batterie;
    }

    public boolean isAmortisseur() {
        return amortisseur;
    }

    public void setAmortisseur(boolean amortisseur) {
        this.amortisseur = amortisseur;
    }

    public boolean isFrein_main() {
        return frein_main;
    }

    public void setFrein_main(boolean frein_main) {
        this.frein_main = frein_main;
    }

    public boolean isFeu_d_eclairage() {
        return feu_d_eclairage;
    }

    public void setFeu_d_eclairage(boolean feu_d_eclairage) {
        this.feu_d_eclairage = feu_d_eclairage;
    }

    public String getAutre() {
        return Autre;
    }

    public void setAutre(String Autre) {
        this.Autre = Autre;
    }

   
   
   
    
   
}

File: src/Entity/Maintenance.php
<?php

namespace App\Entity;

use App\Repository\MaintenanceRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

#[ORM\Entity(repositoryClass: MaintenanceRepository::class)]
class Maintenance
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name:"id_maintenance")]
    #[Groups("maintenance")]
    private ?int $id = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false,name:"id_user",referencedColumnName:"id_user")]
    #[Groups("maintenance")]
    private ?User $user = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Groups("maintenance")]
    private ?\DateTimeInterface $date_maintenance = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $panne_moteur = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $pompe_a_eau = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $patin = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $essuie_glace = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $radiateur = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $ventilateur = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $duride = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $fuite_d_huile = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $vidange = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $filtre = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $batterie = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $amortisseur = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $frein_main = null;

    #[ORM\Column]
    #[Groups("maintenance")]
    private ?bool $feu_d_eclairage = null;

    #[ORM\Column(length:255,nullable:true)]
    private ?string $Autre = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getDateMaintenance(): ?\DateTimeInterface
    {
        return $this->date_maintenance;
    }

    public function setDateMaintenance(\DateTimeInterface $date_maintenance): self
    {
        $this->date_maintenance = $date_maintenance;

        return $this;
    }

    public function isPanneMoteur(): ?bool
    {
        return $this->panne_moteur;
    }

    public function setPanneMoteur(bool $panne_moteur): self
    {
        $this->panne_moteur = $panne_moteur;

        return $this;
    }

    public function isPompeAEau(): ?bool
    {
        return $this->pompe_a_eau;
    }

    public function setPompeAEau(bool $pompe_a_eau): self
    {
        $this->pompe_a_eau = $pompe_a_eau;

        return $this;
    }

    public function isPatin(): ?bool
    {
        return $this->patin;
    }

    public function setPatin(bool $patin): self
    {
        $this->patin = $patin;

        return $this;
    }

    public function isEssuieGlace(): ?bool
    {
        return $this->essuie_glace;
    }

    public function setEssuieGlace(bool $essuie_glace): self
    {
        $this->essuie_glace = $essuie_glace;

        return $this;
    }

    public function isRadiateur(): ?bool
    {
        return $this->radiateur;
    }

    public function setRadiateur(bool $radiateur): self
    {
        $this->radiateur = $radiateur;

        return $this;
    }

    public function isVentilateur(): ?bool
    {
        return $this->ventilateur;
    }

    public function setVentilateur(bool $ventilateur): self
    {
        $this->ventilateur = $ventilateur;

        return $this;
    }

    public function isDuride(): ?bool
    {
        return $this->duride;
    }

    public function setDuride(bool $duride): self
    {
        $this->duride = $duride;

        return $this;
    }

    public function isFuiteDHuile(): ?bool
    {
        return $this->fuite_d_huile;
    }

    public function setFuiteDHuile(bool $fuite_d_huile): self
    {
        $this->fuite_d_huile = $fuite_d_huile;

        return $this;
    }

    public function isVidange(): ?bool
    {
        return $this->vidange;
    }

    public function setVidange(bool $vidange): self
    {
        $this->vidange = $vidange;

        return $this;
    }

    public function isFiltre(): ?bool
    {
        return $this->filtre;
    }

    public function setFiltre(bool $filtre): self
    {
        $this->filtre = $filtre;

        return $this;
    }

    public function isBatterie(): ?bool
    {
        return $this->batterie;
    }

    public function setBatterie(bool $batterie): self
    {
        $this->batterie = $batterie;

        return $this;
    }

    public function isAmortisseur(): ?bool
    {
        return $this->amortisseur;
    }

    public function setAmortisseur(bool $amortisseur): self
    {
        $this->amortisseur = $amortisseur;

        return $this;
    }

    public function isFreinMain(): ?bool
    {
        return $this->frein_main;
    }

    public function setFreinMain(bool $frein_main): self
    {
        $this->frein_main = $frein_main;

        return $this;
    }

    public function isFeuDEclairage(): ?bool
    {
        return $this->feu_d_eclairage;
    }

    public function setFeuDEclairage(bool $feu_d_eclairage): self
    {
        $this->feu_d_eclairage = $feu_d_eclairage;

        return $this;
    }

    public function getAutre(): ?string
    {
        return $this->Autre;
    }

    public function setAutre(string $Autre): self
    {
        $this->Autre = $Autre;

        return $this;
    }
}
